### PR TITLE
create cache_decorator

### PR DESCRIPTION
**Pull request by nrjss67**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cachet = {}
+
+    def wrapper(*args) -> int | str:
```
Comment (vsmutok) [inline app/main.py]: - What if the function takes key arguments?

- Also what if decorator is applied to two different functions that take the same arguments?

File: app/main.py [hunk 2]
```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cachet = {}
```
Comment (vsmutok) [inline app/main.py]:
```suggestion
    cache = {}
```

File: app/main.py (hunk 1 above)
Comment (vsmutok) [inline app/main.py]: use `functools.wraps`

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (nrjss67) [reply app/main.py]: Thank you for your corrections.
Fixed.

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**